torch/_torch_docs.py

```diff
@@ -3160,6 +3160,53 @@ def merge_dicts(*dicts):
 
 """)
 
+add_docstr(torch.inner, r"""
+inner(input, other, *, out=None) -> Tensor
+
+Computes the vector inner product for 1D tensors. For higher dimensions, computes
+a contraction of :attr:`input` and :attr:`other` over their last dimension.
+
+.. note::
+
+    If either :attr:`input` or :attr:`other` is a scalar, the result is equivalent
+    to `torch.mul(input, other)`.
+
+    If both :attr:`input` and :attr:`other` are non-scalars, the size of their last
+    dimensions must match and the result is equivalent to `torch.tensordot(input,
+    other, dims=([-1], [-1]))`
+
+Args:
+    input (Tensor): First input tensor
+    other (Tensor): Second input tensor
+
+Keyword args:
+    out (Tensor, optional): Optional output tensor to write result into. The output
+                            shape is :math:`input.shape[:-1] + other.shape[:-1]`.
+
+Example::
+
+    # Vector inner product
+    >>> torch.inner(torch.tensor([1, 2, 3]), torch.tensor([0, 2, 1]))
+    tensor(7)
+
+    # Multidimensional input tensors
+    >>> torch.inner(torch.randn(2, 3), torch.randn(2, 4, 3))
```

Comment:
Examples using randn are tricky because the original values are unknown. Maybe use arange or construct the tensors explicitly?